src/tools/perf/lib/ucp_tests.cc

```diff
@@ -295,7 +295,7 @@ class ucp_perf_test_runner {
 
         ucp_perf_test_prepare_iov_buffers();
 
-        if (CMD == UCX_PERF_CMD_PUT) {
+        if (length != 0) {
```

Comment:
check CMD not length

Reply:
> check CMD not length

optimization reason?
i.e. `(CMD != PUT)` ? only PUT needs this

Reply:
optimization, CMD!=AM

Reply:
why != AM?
setting psn is required by PUT only

Reply:
it has to be `if (CMD == UCX_PERF_CMD_PUT)` then
right?